acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/slashing/SlashingAcceptanceTest.java

```diff
@@ -0,0 +1,166 @@
+import tech.pegasys.web3signer.core.service.http.Eth2SigningRequestBody;
+import tech.pegasys.web3signer.core.signing.KeyType;
+import tech.pegasys.web3signer.dsl.signer.SignerConfigurationBuilder;
+import tech.pegasys.web3signer.dsl.utils.MetadataFileHelpers;
+import tech.pegasys.web3signer.tests.AcceptanceTestBase;
+
+import java.nio.file.Path;
+import java.security.SecureRandom;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.restassured.response.Response;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class SlashingAcceptanceTest extends AcceptanceTestBase {
+
+  private static final MetadataFileHelpers metadataFileHelpers = new MetadataFileHelpers();
+
+  final BLSKeyPair keyPair = BLSKeyPair.random(new SecureRandom());
+
+  void setupSigner(final Path testDirectory, final boolean enableSlashing) {
+    final SignerConfigurationBuilder builder = new SignerConfigurationBuilder();
+    builder.withMode("eth2");
+    builder.withSlashingEnabled(enableSlashing);
+    builder.withSlashingProtectionDbUsername("postgres");
+    builder.withSlashingProtectionDbPassword("postgres");
+    builder.withKeyStoreDirectory(testDirectory);
+
+    final Path keyConfigFile = testDirectory.resolve("keyfile.yaml");
+    metadataFileHelpers.createUnencryptedYamlFileAt(
+        keyConfigFile, keyPair.getSecretKey().toBytes().toHexString(), KeyType.BLS);
+
+    startSigner(builder.build());
+  }
+
+  @Test
+  void canSignSameAttestationTwiceWhenSlashingIsEnabled(@TempDir Path testDirectory)
+      throws JsonProcessingException {
+
+    setupSigner(testDirectory, true);
+
+    final Eth2SigningRequestBody request =
+        new Eth2SigningRequestBody(
+            Bytes.fromHexString("0x01"),
+            ArtifactType.ATTESTATION,
+            null,
+            UInt64.valueOf(5L),
+            UInt64.valueOf(6L));
+
+    final Response initialResponse =
+        signer.eth2Sign(keyPair.getPublicKey().toBytesCompressed().toHexString(), request);
+    assertThat(initialResponse.getStatusCode()).isEqualTo(200);
+    final Response secondResponse =
+        signer.eth2Sign(keyPair.getPublicKey().toBytesCompressed().toHexString(), request);
+    assertThat(secondResponse.getStatusCode()).isEqualTo(200);
+  }
+
+  @Test
+  void cannotSignASecondAttestationForSameSlotWithDifferentSigningRoot(@TempDir Path testDirectory)
+      throws JsonProcessingException {
+    setupSigner(testDirectory, true);
+
+    final Bytes signingRoot = Bytes.fromHexString("0x01");
+    final Eth2SigningRequestBody initialRequest =
+        new Eth2SigningRequestBody(
+            signingRoot, ArtifactType.ATTESTATION, null, UInt64.valueOf(5L), UInt64.valueOf(6L));
+
+    final Response initialResponse =
+        signer.eth2Sign(keyPair.getPublicKey().toBytesCompressed().toHexString(), initialRequest);
```

Comment:
keyPair.getPublicKey().toBytesCompressed().toHexString() can be shortened to keyPair.getPublicKey.toString()

Reply:
done